### PR TITLE
Updating rule documentation for 2065

### PR DESCRIPTION
**Pull request by dsgouda**
@olydis can you please verify the documentation I have written for this rule.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: documentation/openapi-authoring-automated-guidelines.md [hunk 1]
```diff
@@ -1015,13 +1016,13 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
 
 Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
 
-### <a name="R2064" />R2064 LicenseMissing
-**Output Message**: Please provide correct licensing information here. Acceptable value: "name": "MICROSOFT\_MIT\_NO\_VERSION"
+### <a name="R2065" />R2065 LicenseHeaderMustNotBeSpecified
+**Output Message**: License header must not be specified in the OpenAPI document.
 
-**Description**: Each OpenAPI json document must contain the license object in its `info` section which must set the name to `MICROSOFT\_MIT\_NO\_VERSION`. This is necessary for AutoRest to insert the appropriate headers in the generated code.
+**Description**: `x-ms-code-generation-settings` must not have the header section specified in the OpenAPI documents since each SDK has a different licensing header, this information must be provided either from the command line or the configuration file.
```
Comment (veronicagg) [inline documentation/openapi-authoring-automated-guidelines.md]: it appears that the rules changed quite a bit, could you provide more context on this? can you link to the rule and the original issue and comment on why the change? If the licence header must not be specified, then is it an error now?

File: documentation/openapi-authoring-automated-guidelines.md [hunk 2]
```diff
@@ -1015,13 +1016,13 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
...
 
-**Why the rule is important**: Missing or invalid license information can cause incorrect or empty headers in the generated code.
+**Why the rule is important**: License information must not be specified in the OpenAPI document since different SDKs have different headers.
```
Comment (veronicagg) [inline documentation/openapi-authoring-automated-guidelines.md]: what are the SDKs different headers that you're referring to? is it up to each SDK?

**Review by veronicagg — CHANGES_REQUESTED** [review]

File: documentation/openapi-authoring-automated-guidelines.md (hunk 1 above)
Comment (dsgouda) [inline documentation/openapi-authoring-automated-guidelines.md]: Yes, @olydis I think you can give a better gist here

File: documentation/openapi-authoring-automated-guidelines.md (hunk 2 above)
Comment (dsgouda) [inline documentation/openapi-authoring-automated-guidelines.md]: I believe so. Again, this was per conversations with @olydis

File: documentation/openapi-authoring-automated-guidelines.md (hunk 1 above)
Comment (olydis) [inline documentation/openapi-authoring-automated-guidelines.md]: Original: having `info.license` is cool and all, but it is not used for code gen which this rule sounds like it was for: the `MICROSOFT_MIT_NO_VERSION` constant has no meaning in `info.license`, but in `info.x-ms-code-generation-settings.header` - this leads to the headers that go into all generated code. Now here is the situation: Different SDKs have different demands regarding that, so it really makes zero sense to put something that is not only codegen specific, but even language specific into a Swagger file. I think *most* SDKs ship with `MICROSOFT_MIT_NO_VERSION`, but not all (at least Python I think). In the past, we had people put `MIT` into that field which has no meaning, so in the end *all* SDK teams had fun editing the Swagger before being able to generate code.

TL;DR: this field was a bad idea, causes confusion and pain, doesn't belong in Swagger

File: documentation/openapi-authoring-automated-guidelines.md (hunk 2 above)
Comment (olydis) [inline documentation/openapi-authoring-automated-guidelines.md]: I'd reword that a bit to make clear that it's about the license of *generated* stuff, since the license of the Swagger (`info.license`) is of course something you can specify

File: documentation/openapi-authoring-automated-guidelines.md [hunk 3]
```diff
@@ -1015,13 +1016,13 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
...
+### <a name="R2065" />R2065 LicenseHeaderMustNotBeSpecified
+**Output Message**: License header must not be specified inside x-ms-code-generation settings. This is different for different sdks generated and is passed via command line/config file when generating the sdk.
 
-**Description**: Each OpenAPI json document must contain the license object in its `info` section which must set the name to `MICROSOFT\_MIT\_NO\_VERSION`. This is necessary for AutoRest to insert the appropriate headers in the generated code.
+**Description**: `x-ms-code-generation-settings` must not have the license section specified in the OpenAPI documents since each SDK has a different licensing header, this information must be provided either from the command line or the configuration file when actually generating the sdk.
 
-**Why the rule is important**: Missing or invalid license information can cause incorrect or empty headers in the generated code.
+**Why the rule is important**: License information must not be specified in the `x-ms-code-generation-settings` OpenAPI document since different SDKs have different headers.
```
Comment (olydis) [inline documentation/openapi-authoring-automated-guidelines.md]: > the `x-ms-code-generation-settings` OpenAPI document

feels like it misses a "section of the" or something.

also, rewrite "different SDKs have" to "different SDKs *can* have"

File: documentation/openapi-authoring-automated-guidelines.md [hunk 4]
```diff
@@ -1015,13 +1016,13 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
 
 Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
 
-### <a name="R2064" />R2064 LicenseMissing
-**Output Message**: Please provide correct licensing information here. Acceptable value: "name": "MICROSOFT\_MIT\_NO\_VERSION"
+### <a name="R2065" />R2065 LicenseHeaderMustNotBeSpecified
+**Output Message**: License header must not be specified inside x-ms-code-generation settings. This is different for different sdks generated and is passed via command line/config file when generating the sdk.
```
Comment (olydis) [inline documentation/openapi-authoring-automated-guidelines.md]: - please quote `x-ms-code-generation` consistently
- sdks => SDKs
- "This is different" => "The license can vary for different"
- the sdk => the SDK

File: documentation/openapi-authoring-automated-guidelines.md [hunk 5]
```diff
@@ -1015,13 +1016,13 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
...
 
-**Description**: Each OpenAPI json document must contain the license object in its `info` section which must set the name to `MICROSOFT\_MIT\_NO\_VERSION`. This is necessary for AutoRest to insert the appropriate headers in the generated code.
+**Description**: `x-ms-code-generation-settings` must not have the license section specified in the OpenAPI documents since each SDK has a different licensing header, this information must be provided either from the command line or the configuration file when actually generating the sdk.
```
Comment (olydis) [inline documentation/openapi-authoring-automated-guidelines.md]: - each SDK has => each generated SDK can have
- header, this => header. This

File: documentation/openapi-authoring-automated-guidelines.md [hunk 6]
```diff
@@ -1015,13 +1016,13 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
...
 
-**Why the rule is important**: Missing or invalid license information can cause incorrect or empty headers in the generated code.
+**Why the rule is important**: License information must not be specified in the `x-ms-code-generation-settings` OpenAPI document since different SDKs have different headers.
 
-**How to fix the violation**: Ensure the `info` section of OpenAPI document has the name property set to `MICROSOFT\_MIT\_NO\_VERSION`.
+**How to fix the violation**: Ensure the `x-ms-code-generation-settings` does not have `header` section.
```
Comment (olydis) [inline documentation/openapi-authoring-automated-guidelines.md]: `x-ms-code-generation-settings` is the section, `header` is just a property.

File: documentation/openapi-authoring-automated-guidelines.md (hunk 5 above)
Comment (olydis) [inline documentation/openapi-authoring-automated-guidelines.md]: "Output Message" and "Description" are identical modulo spelling and grammar used. If there is really not more to say in the description than in the output message, just copy paste the output message, I see no value in having identical information in similar rather than identical, but high quality words. 😉

File: documentation/openapi-authoring-automated-guidelines.md [hunk 7]
```diff
@@ -115,6 +115,7 @@ We request OpenAPI(Swagger) spec authoring be assigned to engineers who have an
 | [R1006](#R1006)	| [PutInOperationName](#R1006) | 'PUT' operation '{0}' should use method name 'Create'. Note: If you have already shipped an SDK on top of this spec, fixing this warning may introduce a breaking change. | Warning |
 | [R1007](#R1007) | [PatchInOperationName](#R1007) | 'PATCH' operation '{0}' should use method name 'Update'. Note: If you have already shipped an SDK on top of this spec, fixing this warning may introduce a breaking change. | Warning |
 | [R1011](#R1011) | [HttpsSupportedScheme](#R1011) | 'Azure Resource Management only supports HTTPS scheme. | Warning |
+| [R2065](#R2065) | [LicenseHeaderMustNotBeSpecified](#R2065) | License header must not be specified inside `x-ms-code-generation` settings of OpenAPI document. | Warning |
```
Comment (olydis) [inline documentation/openapi-authoring-automated-guidelines.md]: there is no `x-ms-code-generation`, only `x-ms-code-generation-settings`

**Review by olydis — CHANGES_REQUESTED** [review]

File: documentation/openapi-authoring-automated-guidelines.md (hunk 4 above)
Comment (dsgouda) [inline documentation/openapi-authoring-automated-guidelines.md]: Output messages cannot have the tildes, will address the rest

**Comment by salameer** [comment]
Folks let's wrap up on this :)

File: documentation/openapi-authoring-automated-guidelines.md [hunk 8]
```diff
@@ -1030,13 +1031,11 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
 Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
 
 ### <a name="R2065" />R2065 LicenseHeaderMustNotBeSpecified
-**Output Message**: License header must not be specified in the OpenAPI document.
+**Output Message**: License header must not be specified inside x-ms-code-generation-settings. The license can vary for different SDKs generated and is passed via command line/config file when generating the SDK.
```
Comment (veronicagg) [inline documentation/openapi-authoring-automated-guidelines.md]: could you add a link to a doc explaining how to add this to the config file?

File: documentation/openapi-authoring-automated-guidelines.md [hunk 9]
```diff
@@ -1030,13 +1031,11 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
...
 
-**Description**: `x-ms-code-generation-settings` must not have the header section specified in the OpenAPI documents since each SDK has a different licensing header, this information must be provided either from the command line or the configuration file.
+**Description**: `x-ms-code-generation-settings` must not have the license section specified in the OpenAPI documents since each generated SDK can have a different licensing header. This information must be provided either from the command line or the configuration file when actually generating the sdk.
```
Comment (veronicagg) [inline documentation/openapi-authoring-automated-guidelines.md]: let's add a link to documentation on how to add this to the config file.

**Review by veronicagg — APPROVED** [review]
minor comment, rest looks good.

**Review by olydis — APPROVED** [review]
